src/runtime/hexagon_remote/halide_host.cpp

```diff
@@ -1,23 +1,237 @@
-#include <rpcmem.h>
+#include <pthread.h>
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <linux/types.h>
+#include <android/log.h>
+
+#include "bin/src/halide_hexagon_remote.h"
+
+typedef halide_hexagon_remote_handle_t handle_t;
+typedef halide_hexagon_remote_buffer buffer;
+
+typedef int ion_user_handle_t;
```

Comment:
I assume these are mirroring data structures from Ion. Can we add a comment explaining this and perhaps what Ion is?